extensions/data-plane/data-plane-aws-s3/src/test/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSinkTest.java

```diff
@@ -107,6 +114,60 @@ void transferParts_multiPart_succeeds(List inputStream) {
         assertThat(completeMultipartUploadRequest.multipartUpload().parts()).hasSize(2);
     }
 
+    @ParameterizedTest
+    @ArgumentsSource(MultiPartsInputs.class)
+    void transferParts_fails_at_datasink(List inputStream) {
+        var s3DatasinkS3Client = mock(S3Client.class);
+
+        var s3Datasink = S3DataSink.Builder.newInstance()
+                .bucketName(BUCKET_NAME)
+                .keyName(KEY_NAME)
+                .client(s3DatasinkS3Client)
+                .requestId(TestFunctions.createRequest(S3BucketSchema.TYPE).build().getId())
+                .executorService(Executors.newFixedThreadPool(2))
+                .monitor(mock(Monitor.class))
+                .chunkSizeBytes(CHUNK_SIZE_BYTES)
+                .build();
+
+        when(s3DatasinkS3Client.createMultipartUpload(any(CreateMultipartUploadRequest.class)))
+                .thenThrow(new RuntimeException(ERROR_MESSAGE));
+
+        var result = s3Datasink.transferParts(inputStream);
+
+        var expectedMessage = format("Error transferring the %s object: Fail uploading object to the destination bucket: %s",
+                KEY_NAME, ERROR_MESSAGE);
+
+        assertThat(result.failed()).isTrue();
+        assertThat(result.getFailureDetail()).isEqualTo(expectedMessage);
+
+    }
+
+    @Test
+    void transferParts_fails_at_datasource() {
+        var s3DatasourceS3Client = mock(S3Client.class);
+
+        var s3Datasource = S3DataSource.Builder.newInstance()
+                .keyName(KEY_NAME)
+                .bucketName(BUCKET_NAME)
+                .client(s3DatasourceS3Client).build();
+
+        when(s3DatasourceS3Client.getObject(any(GetObjectRequest.class)))
+                .thenThrow(new RuntimeException(ERROR_MESSAGE));
+
+        var results = PartitionIterator.streamOf(s3Datasource.openPartStream().getContent(), 5)
+                .map(parts -> dataSink.transferParts(parts)).toList();
```

Comment:
the scope of this test looks too broad, if it involves `PartitionIterator` and `dataSink`, likely this can be considered a component/integration test, and it should be located elsewhere.

Reply:
Removed in refactored code.